### PR TITLE
fix: now explore link is actively styled when neccessary

### LINKED ISSUE #2336: Bug: Explore navigation link does not retain it's current(`active`) page style
### Describe the bug

Highlight top navigation link does not retain it's current page style

### Steps to reproduce

1. go to any page in the app.
2. Click the Explore link in the top navigation.
3. Notice it does'n get set to orange for the current page style.

CleanShot 2023-12-13 at 18 16 10

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR fixes the issue with the explore nav link active style. We might want to consider updating the regex to be more specific in the future as we build more routes, but for now i think this covers our use case.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
fixes #2336 

### Before
https://github.com/open-sauced/app/assets/62995161/29812f2b-6b24-40c8-b4f0-fbb6461b980d

### After
https://github.com/open-sauced/app/assets/62995161/30a2ce67-8873-4dc1-bd79-7586903b520d

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by nickytonline** [comment]
If I navigate to the Explore link I get a 404. The URL doesn't look right. It's http://localhost:3000/dashboard/filter/recent.  Not sure what changed in the PR to cause that. This seems to be the only thing related to Hacktoberfest that changed.

`  const userInterest = gitHubUser?.interests.split(",")[0] ?? "hacktoberfest";`

CleanShot 2023-12-18 at 09 30 36

On the beta branch it loads a hacktoberfest URL, http://localhost:3000/hacktoberfest/dashboard/filter/recent?range=30

**Comment by OgDev-01** [comment]
> If I navigate to the Explore link I get a 404. The URL doesn't look right. It's http://localhost:3000/dashboard/filter/recent. Not sure what changed in the PR to cause that. This seems to be the only thing related to Hacktoberfest that changed.

I tried to reproduce this on the preview link, and it works as expected 🤔. Have you tried incognito mode just to get on a clean slate without cache?

**Comment by nickytonline** [comment]
> deploy-preview-2364--oss-insights.netlify.app

It happens on the deploy preview as well. It only happens if you're logged in.

**Logged Out**

CleanShot 2023-12-18 at 10 10 51

**Logged In**

CleanShot 2023-12-18 at 10 09 50

**Comment by OgDev-01** [comment]
Here is what i get from my end
https://github.com/open-sauced/app/assets/62995161/e9a0517c-b015-4ecf-a99f-613134875db0

 We could do with some help from @open-sauced/engineering to test run this to be sure there is an issue here before digging into fixes.

**Comment by a0m0rajab** [comment]
I tested it on deploy preview and it works smoothly on my side.

**Comment by nickytonline** [comment]
> I tested it on deploy preview and it works smoothly on my side. @nickytonline What are the steps you followed that caused this issue?

All I did was test it logged out (all working) and then I logged in and it gave a 404 for the explore page as the URL is incorrect.

**Comment by a0m0rajab** [comment]
I could not reproduce the issue/

**Comment by nickytonline** [comment]
I guess see if anyone else runs into the issue. @jpmcb or @brandonroberts, do you exhibit the same behaviour I do on the deploy preview?

**Comment by BekahHW** [comment]
The deploy preview also works fine for me.

**Comment by nickytonline** [comment]
> The deploy preview also works fine for me.

Clearly it's just me lol then.

**Comment by nickytonline** [comment]
I can consistently reproduce it and I've even cleared cookies and all app data in the browser. So I don't know if it's my user in beta or what. 🙃

**Comment by brandonroberts** [comment]
@nickytonline you're correct `/dashboard/filter/recent` is invalid. It should have a topic first, such as `/javascript/dashboard/filter/recent`

**Comment by brandonroberts** [comment]
I can see the 404 using the developer preview also without the topic

**Comment by brandonroberts** [comment]
This also makes the `Explore` link active when on the `activity` tab for a list, which is not intended.

image

**Comment by OgDev-01** [comment]
> I can see the 404 using the developer preview also without the topic

I'm lost here 😀... how is the redirect taking the user to `/dashboard/filter/recent` without a topic on login 🤔🤔?. from the implementation, javascript is set to be the default fallback

**Comment by OgDev-01** [comment]
> This also makes the `Explore` link active when on the `activity` tab for a list, which is not intended.
> 
> image

This has been escaped now with an updated regex @brandonroberts.

File: components/organisms/TopNav/top-nav.tsx [hunk 1]
```diff
@@ -64,7 +66,7 @@ const Nav = ({ className, name = "Main" }: { className?: string; name?: string }
         </li>
         <li>
           <Link
-            className={`text-sm ${getActiveStyle(router.asPath === `/${userInterest}/dashboard/filter/recent`)}`}
+            className={`text-sm ${getActiveStyle(explorePageUrlPattern.test(router.asPath))}`}
```
Comment (nickytonline) [inline components/organisms/TopNav/top-nav.tsx:69]: Wouldn't the regex just need to match the pattern in the url below? `/${userInterest}/dashboard/filter/recent`? So something like `\/[^\/]+\/dashboard\/filter\/recent$` for the path?

See the [explainer of the above regex](https://regex101.com/?regex=%5B%5Cp%7BLl%7D0-9-%5D+) on regex101.

Comment (OgDev-01) [reply components/organisms/TopNav/top-nav.tsx:69]: Nice thoughts 👍 

The changes I made were accounting for all the tab pages around the explore, e.g `reports`, `contributors`, and `activity`. I feel it is ideal to keep the active state on those pages as well. thoughts @open-sauced/engineering?

**Review by brandonroberts — APPROVED** [review]

File: components/organisms/TopNav/top-nav.tsx (hunk 1 above)
Comment (nickytonline) [reply components/organisms/TopNav/top-nav.tsx:69]: Post PTO, we'll go with what you have proposed as I was suggesting a regex based on what I believe wasn't working prior to this fix.

```
className={`text-sm ${getActiveStyle(router.asPath === `/${userInterest}/dashboard/filter/recent`)}`}
```

**Review by nickytonline — APPROVED** [review]
:shipit:

**Pull request merged**